src/main/resources/sql/create.sql

```diff
@@ -0,0 +1,40 @@
+CREATE TABLE IF NOT EXISTS "shop" (
+  "id"      SERIAL PRIMARY KEY,
+  "name"    VARCHAR(255)        NOT NULL,
+  "address" VARCHAR(255) UNIQUE NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS "customer" (
+  "id"      SERIAL PRIMARY KEY,
+  "name"    VARCHAR(255) NOT NULL,
+  "surname" VARCHAR(255) NOT NULL,
+  "age"     SMALLINT     NOT NULL,
+  "shop_id" INTEGER      NOT NULL,
+  FOREIGN KEY ("shop_id") REFERENCES "shop" ("id")
+);
+
+CREATE TABLE IF NOT EXISTS "purchase" (
+  "id"          SERIAL PRIMARY KEY,
+  "date"        TIMESTAMP NOT NULL,
+  "customer_id" INTEGER   NOT NULL,
+  UNIQUE ("date", "customer_id"),
+  FOREIGN KEY ("customer_id") REFERENCES "customer" ("id")
+);
+
+CREATE TABLE IF NOT EXISTS "product" (
+  "id"    SERIAL PRIMARY KEY,
+  "name"  VARCHAR(255) NOT NULL,
+  "price" MONEY        NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS "purchase_detail" (
+  "id"          SERIAL PRIMARY KEY,
+  "quantity"    INT2    NOT NULL,
+  "price"       MONEY   NOT NULL,
+  "shop_id"     INTEGER NOT NULL,
+  "purchase_id" INTEGER NOT NULL,
+  "product_id"  INTEGER NOT NULL,
+  FOREIGN KEY ("product_id") REFERENCES "product" ("id"),
+  FOREIGN KEY ("purchase_id") REFERENCES "purchase" ("id"),
+  FOREIGN KEY ("shop_id") REFERENCES "shop" ("id")
```

Comment:
What if the record from "purchase" table will be removed?
Try use ON DELETE CASCADE.

> ON DELETE CASCADE option is to specify whether you want rows deleted in a child table when corresponding rows are deleted in the parent table. If you do not specify cascading deletes, the default behaviour of the database server prevents you from deleting data in a table if other tables reference it.